edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Kubernetes.IntegrationTest/EdgeDeploymentControllerTest.cs

```diff
@@ -0,0 +1,590 @@
+        }
+
+        [Fact]
+        public async Task CheckIfCreateDeploymentIsSuccessfulWithNoResources()
+        {
+            var moduleName = "module-a";
+            var deviceSelector = $"{Kubernetes.Constants.K8sEdgeDeviceLabel}=deviceid,{Kubernetes.Constants.K8sEdgeHubNameLabel}=hostname";
+            var moduleLifeCycleManager = new DummyModuleIdentityLifecycleManager(
+                "hostname",
+                "gatewayhostname",
+                "deviceid",
+                moduleName,
+                new ConnectionStringCredentials("connectionString"));
+            var creatOptions = CreatePodParameters.Create();
+            var controller = this.CreateDeploymentController(deviceSelector, moduleLifeCycleManager, string.Empty, string.Empty);
+            string imageName = "test-image:1";
+            KubernetesModule km1 = this.CreateKubernetesModule(moduleName, creatOptions, imageName);
+
+            moduleLifeCycleManager.SetModules(moduleName);
+            await controller.DeployModulesAsync(ModuleSet.Create(km1), ModuleSet.Empty);
+
+            V1DeploymentList currentDeployments = await this.client.ListDeployments(deviceSelector);
+            V1ServiceAccountList currentServiceAccounts = await this.client.ListServiceAccounts(deviceSelector);
+            V1ServiceList currentServices = await this.client.ListServices(deviceSelector);
+            V1PersistentVolumeClaimList currentPvcList = await this.client.ListPeristentVolumeClaims();
+            Assert.Single(currentDeployments.Items, d => d.Metadata.Name == moduleName);
+            Assert.Single(currentServiceAccounts.Items, sa => sa.Metadata.Name == moduleName);
+            Assert.Empty(currentServices.Items);
+            Assert.Empty(currentPvcList.Items);
+        }
+
+        [Fact]
+        public async Task CheckIfCreateDeploymentNoServiceWithPvcIsSuccessful()
+        {
+            var moduleName = "module-a";
+            var deviceSelector = $"{Kubernetes.Constants.K8sEdgeDeviceLabel}=deviceid,{Kubernetes.Constants.K8sEdgeHubNameLabel}=hostname";
+            var moduleLifeCycleManager = new DummyModuleIdentityLifecycleManager(
+                "hostname",
+                "gatewayhostname",
+                "deviceid",
+                moduleName,
+                new ConnectionStringCredentials("connectionString"));
+            var hostConfigs = new HostConfig
+            {
+                Mounts = new List<Mount>
+                {
+                    new Mount
+                    {
+                        Type = "volume",
+                        ReadOnly = true,
+                        Source = "pvname",
+                        Target = "/tmp/volume"
+                    }
+                }
+            };
+            var persistentVolumeName = "pvname";
+            var creatOptions = new CreatePodParameters(null, null, hostConfigs, null, null);
```

Comment:
CreatePodParameters has Create() factory method that you can use instead of using null explicitly